src/FakturoidManager.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace Fakturoid;
+
+use Fakturoid\Auth\AuthProvider;
+use Fakturoid\Auth\CredentialCallback;
+use Fakturoid\Auth\Credentials;
+use Fakturoid\Enum\AuthTypeEnum;
+use Fakturoid\Exception\AuthorizationFailedException;
+use Fakturoid\Provider\AccountProvider;
+use Fakturoid\Provider\BankAccountProvider;
+use Fakturoid\Provider\EventProvider;
+use Fakturoid\Provider\ExpenseProvider;
+use Fakturoid\Provider\GeneratorProvider;
+use Fakturoid\Provider\InboxFileProvider;
+use Fakturoid\Provider\InventoryItemProvider;
+use Fakturoid\Provider\InventoryMoveProvider;
+use Fakturoid\Provider\InvoiceProvider;
+use Fakturoid\Provider\NumberFormatProvider;
+use Fakturoid\Provider\RecurringGeneratorProvider;
+use Fakturoid\Provider\SubjectProvider;
+use Fakturoid\Provider\TodoProvider;
+use Fakturoid\Provider\UserProvider;
+use Psr\Http\Client\ClientInterface;
+
+class FakturoidManager
+{
+    private readonly AuthProvider $authProvider;
+    private readonly Dispatcher $dispatcher;
+    private readonly AccountProvider $accountProvider;
+    private readonly BankAccountProvider $bankAccountProvider;
+    private readonly EventProvider $eventProvider;
+    private readonly ExpenseProvider $expenseProvider;
+    private readonly GeneratorProvider $generatorProvider;
+    private readonly InboxFileProvider $inboxFileProvider;
+    private readonly InventoryItemProvider $inventoryItemsProvider;
+    private readonly InventoryMoveProvider $inventoryMoveProvider;
+    private readonly InvoiceProvider $invoiceProvider;
+    private readonly NumberFormatProvider $numberFormatProvider;
+    private readonly RecurringGeneratorProvider $recurringGeneratorProvider;
+    private readonly SubjectProvider $subjectProvider;
+    private readonly TodoProvider $todoProvider;
+    private readonly UserProvider $userProvider;
+
+    public function __construct(
+        ClientInterface $client,
+        string $clientId,
+        string $clientSecret,
+        string $userAgent,
+        ?string $accountSlug = null,
+        ?string $redirectUri = null
+    ) {
+        $this->authProvider = new AuthProvider($clientId, $clientSecret, $redirectUri, $client);
+        $this->dispatcher = new Dispatcher($userAgent, $this->authProvider, $client, $accountSlug);
+
+        $this->accountProvider = new AccountProvider($this->dispatcher);
+        $this->bankAccountProvider = new BankAccountProvider($this->dispatcher);
+        $this->eventProvider = new EventProvider($this->dispatcher);
+        $this->expenseProvider = new ExpenseProvider($this->dispatcher);
+        $this->generatorProvider = new GeneratorProvider($this->dispatcher);
+        $this->inboxFileProvider = new InboxFileProvider($this->dispatcher);
+        $this->inventoryItemsProvider = new InventoryItemProvider($this->dispatcher);
```

Comment:
Tady jsem si ještě všim plurálu. Hele jestli by ti přišlo přirozenější mít všude plurál, tak já bych byl taky pro.

Reply:
No všude být asi nemá. Když koukám, tak je to skoro všude krom account

Reply:
a u `user` je to takové matoucí.. ale tam asi by to mohlo být v plurálu

Reply:
V dokumentaci koukám že máme taky jen Account jako singulár a zbytek plurál, tak to můžeme udělat stejně, jestli to tady dává smysl. Users je takovej divnej, protože to je mix 2 endpointů, který spolu nesouvisej moc, ale s tím můžeme žít.

Reply:
Tak u `User` chceme plurál nebo ne?

Reply:
můžem u něj dát plurál

Reply:
@ollie hotovo. Upravil jsem to v readme, property i gettery